compiler/rustc_infer/src/traits/project.rs

```diff
@@ -78,11 +78,12 @@ pub struct ProjectionCacheStorage<'tcx> {
 #[derive(Copy, Clone, Debug, Hash, PartialEq, Eq)]
 pub struct ProjectionCacheKey<'tcx> {
     ty: ty::AliasTy<'tcx>,
+    param_env: ty::ParamEnv<'tcx>,
```

Comment:
won't that lead to tons of cache misses?

Reply:
Almost all code that isn't the `AutoTraitFinder` will always use the same `ParamEnv` for a single `InferCtxt`, so I don't expect there to be many cache misses if any at all. But maybe start a perf run just to confirm? (I don't have permission)